static/app/views/issueDetails/activitySection.tsx

```diff
@@ -0,0 +1,152 @@
+  placeholderText: string;
+  addMutationOptions?: MutateOptions<TData, TError, TVariables, TContext>;
+  deleteMutationOptions?: MutateOptions<TData, TError, TVariables, TContext>;
+  updateMutationOptions?: MutateOptions<TData, TError, TVariables, TContext>;
+};
+
+function ActivitySection(props: Props) {
+  const {
+    group,
+    placeholderText,
+    mutators,
+    updateMutationOptions,
+    deleteMutationOptions,
+    addMutationOptions,
+  } = props;
+  const organization = useOrganization();
+  const {addComment, updateComment, deleteComment} = mutators;
+
+  const [inputId, setInputId] = useState(uniqueId());
+
+  const me = ConfigStore.get('user');
+  const projectSlugs = group && group.project ? [group.project.slug] : [];
+  const noteProps = {
+    minHeight: 140,
+    group,
+    projectSlugs,
+    placeholder: placeholderText,
+  };
+
+  return (
+    <Fragment>
+      <ActivityItem noPadding author={{type: 'user', user: me}}>
+        <NoteInputWithStorage
+          key={inputId}
+          storageKey="groupinput:latest"
+          itemKey={group.id}
+          onCreate={n => {
+            const newActivity: GroupActivityNote = {
+              id: uniqueId(), // temporary unique id, for cache use only
+              data: n,
+              type: GroupActivityType.NOTE,
+              dateCreated: new Date().toISOString(),
+              project: group.project,
+              user: me,
+            };
+            addComment(n, [newActivity, ...group.activity], addMutationOptions);
+            setInputId(uniqueId());
+          }}
+          {...noteProps}
+        />
+      </ActivityItem>
+
+      {group.activity.map(item => {
+        const authorName = item.user ? item.user.name : 'Sentry';
+
+        if (item.type === GroupActivityType.NOTE) {
+          return (
+            <ErrorBoundary mini key={`note-${item.id}`}>
+              <Note
+                showTime={false}
+                text={item.data.text}
+                noteId={item.id}
+                user={item.user as User}
+                dateCreated={item.dateCreated}
+                authorName={authorName}
+                onDelete={() => {
+                  // @ts-ignore
+                  if (group.issueCategory !== 'feedback') {
+                    const restore = group.activity.find(
+                      activity => activity.id === item.id
+                    );
+                    const index = GroupStore.removeActivity(group.id, item.id);
+
+                    if (index === -1 || restore === undefined) {
+                      addErrorMessage(t('Failed to delete comment'));
+                      return;
+                    }
+                  }
+                  deleteComment(
+                    item.id,
+                    group.activity.filter(a => a.id !== item.id),
+                    deleteMutationOptions
+                  );
+                }}
+                onUpdate={n => {
+                  item.data.text = n.text;
```

Comment:
this was probably there before, but its weird

Reply:
this is something i added actually😅 mainly for feedback to optimistically update the text after it's edited

Reply:
how does that work? Wouldn't item come from a state somewhere, feels like mutating it wouldn't work